src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8931,9 +8933,11 @@ private bool IsAwaitExpression()
                 // If we see an await followed by a token that cannot follow an identifier, parse await as a unop.
                 // BindAwait() catches the cases where await successfully parses as a unop but is not in an async
                 // function, and reports an appropriate ERR_BadAwaitWithoutAsync* error.
-                switch (this.PeekToken(1).Kind)
+                var next = PeekToken(1);
+                switch (next.Kind)
                 {
                     case SyntaxKind.IdentifierToken:
+                        return next.ContextualKind != SyntaxKind.WithKeyword;
```

Comment:
>ContextualKind != SyntaxKind.WithKeyword [](start = 36, length = 40)

This would break an existing program declaring a local `await with;` where `await` is the name of the type and `with` is the name of the local.

Reply:
I don't understand how. Wouldn't that be a local declaration, not an expression? So we wouldn't get here in the first place?

Reply:
But you could have that in an expression as well, right? `M(out await with)`

Reply:
I suspect we call this when attempting to parse an expression statement.

---
In reply to: [390667442](https://github.com/dotnet/roslyn/pull/42304#discussion_r390667442) [](ancestors = 390667442)

Reply:
The answer is no, this function is never called in either of those cases. Parsing an expression statement only happens if parsing as a local declaration fails, and parsing as an `out` expression is handled separately.